sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -500,6 +504,8 @@ case class Range(
   extends LeafNode with MultiInstanceRelation {
 
   require(step != 0, s"step ($step) cannot be 0")
+  require(start != end, s"start ($start) cannot be equal to end ($end)")
+  require(start < end ^ step < 0, s"the sign of step ($step) is invalid for range ($start, $end)")
```

Comment:
If we allow to return an empty relation/empty range in Range.apply, why we won't allow it too here? Although it's no-op, but looks like it's legal query?